### PR TITLE
Helidon SE client generator

### PR DESCRIPTION
**Pull request by tjquinno**
Basic SE client generation. Resolves oracle/helidon#4265

Also:
* Declares the already-added `helidonVersion` property as a CLI option so it appears in the generated help, including
   * the source change to `JavaHelidonCommonCodegen`
   *  the new version of the generated client and server `.md` documents
* Slightly updates the help text for the client generator so it no longer refers only to MP.

This PR includes many templates in the `client/se` directory that we probably do not need. That will be cleaned up in a subsequent PR.

The OpenAPI document contains all the information we need to generate immediately useful wrapper methods for the remote API invocations. So the SE client generator creates concrete classes (which can be but do not need to be extended).

The generated `ApiClient` class has a `Builder` which encapsulates but exposes the Helidon `WebClient.Builder`, creating a default one according to the information in the OpenAPI document but allowing the developer to augment or completely override it. The resulting `ApiClient` instance has the built `WebClient`.

Each generated `xxxApi` class has a constructor which accepts the `ApiClient` it should use. Each operation from the OpenAPI document has three generated methods: 
* `public Single<WebClientResponse> _operationId_(_arguments_)`
* `protected WebClientRequestBuilder _operationId_RequestBuilder(_arguments_)`
* `protected Single<WebClientResponse> _operationId_Submit(WebClientRequestBuilder, _arguments_)`

The first method simply delegates to the other two. The developer can extend the generated class and override any of these, but most notably the last two to customize their behavior. We expose two extension points so the developer can separately customize how the `WebClientRequestBuilder` is prepared and how the request is submitted (which is where the Helidon web client API accepts the entity/payload).

This approach--an `ApiClient` class and one for each of the APIs--is common across most/all client-side generators. 

The public generated API methods return `Single<WebClientResponse>`, at least as a starting point. The Helidon doc and Slack channels contain much information about how to use the `WebClient`  API. We might decide to generate synchronous methods as well which could return `WebClientResponse` and/or the method's return type if we think it's worth it.

### CODE REVIEW

**Review by tvallin — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/java-helidon/client/libraries/se/beanValidationCore.mustache [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{{#pattern}} @Pattern(regexp="{{{.}}}"){{/pattern}}{{!
```
Comment (spericas) [inline modules/openapi-generator/src/main/resources/java-helidon/client/libraries/se/beanValidationCore.mustache:1]: Do we need any of these BV templates in SE?

Comment (tjquinno) [reply modules/openapi-generator/src/main/resources/java-helidon/client/libraries/se/beanValidationCore.mustache:1]: Most likely not. There are sure to be many templates there now that we will not need. I am trying to get a working starting point for SE client support in our repo with clean-up to follow. I can remove those templates now for sure.

Comment (tjquinno) [reply modules/openapi-generator/src/main/resources/java-helidon/client/libraries/se/beanValidationCore.mustache]: Removed

File: docs/generators/java-helidon-server.md [hunk 2]
```diff
@@ -92,12 +92,14 @@ These options may be applied as additional-properties (cli) or configOptions (pl
...
 |BigDecimal|java.math.BigDecimal|
 |Date|java.util.Date|
+|DateTime|org.joda.time.*|
 |File|java.io.File|
 |HashMap|java.util.HashMap|
 |LinkedHashSet|java.util.LinkedHashSet|
 |List|java.util.*|
-|LocalDate|java.time.LocalDate|
-|OffsetDateTime|java.time.OffsetDateTime|
+|LocalDate|org.joda.time.*|
```
[the remark this answers is not in the document]
Comment (aserkes) [reply docs/generators/java-helidon-server.md:100]: We do not use `joda` but use `java.time`, so I think it has to be changed.

**Review by aserkes — APPROVED** [review]

**Pull request merged**